File: Year-2023/LeetCode-October-Challenge/RemoveColoredPiecesifBothNeighborsaretheSameColor.java
class Solution {
    public boolean winnerOfGame(String colors) {
        int aliceCount = 0, bobCount = 0;
        for(int i = 1; i < colors.length() - 1; i++){
            if(colors.charAt(i - 1) == 'A' && colors.charAt(i) == 'A' && colors.charAt(i + 1) == 'A') aliceCount++;
            if(colors.charAt(i - 1) == 'B' && colors.charAt(i) == 'B' && colors.charAt(i + 1) == 'B') bobCount++;
        }
        return aliceCount > bobCount;
    }
}
public class RemoveColoredPiecesifBothNeighborsaretheSameColor{
    public static void main(String[] args) {
        String colors = "AAABABB";
        System.out.println(new Solution().winnerOfGame(colors));
    }
}

File: Year-2023/LeetCode-October-Challenge/RemoveColoredPiecesifBothNeighborsaretheSameColor.js
var winnerOfGame = function(colors) {
    let aliceCount = 0, bobCount = 0;

    for(let i = 1; i < colors.length - 1; ++i){
        if(colors[i - 1] == 'A' && colors[i] == 'A' && colors[i + 1] == 'A') aliceCount++;
        if(colors[i - 1] == 'B' && colors[i] == 'B' && colors[i + 1] == 'B') bobCount++;
    }
    return aliceCount > bobCount;
};

let colors = "AAABABB";
console.log(winnerOfGame(colors));